### PR TITLE
github_runner_matrix: remove `GITHUB_RUN_ATTEMPT`

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This should no longer be needed after Homebrew/ci-orchestrator@845d26ccc03ffebb06c39d0b9ce81134e8a6688b.

### CODE REVIEW

**Comment by Bo98** [comment]
Oh right this is in a different repo now so we can't really test. Can you try change `.github/workflows/doctor.yml` so we can confirm it works there?

**Comment by carlocab** [comment]
> Oh right this is in a different repo now so we can't really test.

We can, it just requires some workflow file changes.

> Can you try remove it from `.github/workflows/doctor.yml` too so we can confirm it works there?

👍

File: .github/workflows/doctor.yml [hunk 1]
```diff
@@ -18,13 +18,13 @@ jobs:
...
-          - runner: "12-${{github.run_id}}-${{github.run_attempt}}"
+          - runner: "13-arm64-${{ github.run_id }}"
+          - runner: "12-arm64-${{ github.run_id }}"
+          - runner: "12-${{ github.run_id }}"
           - runner: "11-arm64"
             cleanup: true
-          - runner: "11-${{github.run_id}}-${{github.run_attempt}}"
-          - runner: "10.15-${{github.run_id}}-${{github.run_attempt}}"
+          - runner: "11-${{ github.run_id }}"
+          - runner: "10.15-${{ github.run_id }}"
```
Comment (carlocab) [inline .github/workflows/doctor.yml:27]: Does this even still work?

Comment (Bo98) [reply .github/workflows/doctor.yml:27]: It likely does. The image still exists on our Orka cluster and the mapping still exists in ci-orchestrator - there's not really been a need to remove it.

Though we probably should drop it from here anyway.

Comment (carlocab) [reply .github/workflows/doctor.yml:27]: So it does. It needs to be replaced with an Intel Ventura run anyway.

Comment (carlocab) [reply .github/workflows/doctor.yml:27]: ...wait, shouldn't `brew doctor` fail on 10.15 now? With some sort of `unsupported macOS` error.

Comment (Bo98) [reply .github/workflows/doctor.yml:27]: `HOMEBREW_DEVELOPER=1` actually skips that check.

Comment (carlocab) [reply .github/workflows/doctor.yml]: Removed.

Will give this a re-run to see how the orchestrator handles it after.

Comment (Bo98) [reply .github/workflows/doctor.yml]: Good idea :+1:

Comment (carlocab) [reply .github/workflows/doctor.yml]: Rerun attempted... https://github.com/Homebrew/brew/actions/runs/5112934741/jobs/9191889634 (only Intel Monterey)

Comment (carlocab) [reply .github/workflows/doctor.yml]: That works, I think?

Screenshot 2023-05-29 at 23 01 42

**Review by MikeMcQuaid — APPROVED** [review]

File: .github/workflows/doctor.yml (hunk 1 above)
Comment (Bo98) [reply .github/workflows/doctor.yml]: LGTM

Comment (carlocab) [reply .github/workflows/doctor.yml]: Also now running. Nice. Let's try re-running all of them...

**Review by Bo98 — APPROVED** [review]

**Pull request merged**